app/routers/event.py

```diff
@@ -70,93 +76,117 @@ async def create_event_api(event: EventModel, session=Depends(get_db)):
     content = data['description']
-    start = dt.strptime(data['start_date'] + ' ' +
-                        data['start_time'], TIME_FORMAT)
-    end = dt.strptime(data['end_date'] + ' ' + data['end_time'],
-                      TIME_FORMAT)
+    start = dt.strptime(data['start_date'] + ' ' + data['start_time'],
+                        TIME_FORMAT)
+    end = dt.strptime(data['end_date'] + ' ' + data['end_time'], TIME_FORMAT)
     owner_id = get_current_user(session).id
     availability = data.get('availability', 'True') == 'True'
     location = data['location']
     vc_link = data['vc_link']
     category_id = data.get('category_id')
 
     invited_emails = get_invited_emails(data['invited'])
-    uninvited_contacts = get_uninvited_regular_emails(session, owner_id,
-                                                      title, invited_emails)
+    uninvited_contacts = get_uninvited_regular_emails(session, owner_id, title,
+                                                      invited_emails)
 
     if vc_link is not None:
         raise_if_zoom_link_invalid(vc_link)
 
-    event = create_event(session, title, start, end, owner_id, content,
-                         location, vc_link, invitees=invited_emails,
+    event = create_event(session,
+                         title,
+                         start,
+                         end,
+                         owner_id,
+                         content,
+                         location,
+                         vc_link,
+                         invitees=invited_emails,
                          category_id=category_id,
                          availability=availability)
 
-    messages = get_messages(session, event, uninvited_contacts)
-    return RedirectResponse(router.url_path_for('eventview', event_id=event.id)
-                            + f'messages={"---".join(messages)}',
-                            status_code=status.HTTP_302_FOUND)
-
+    if event_img:
+        image = process_image(event_img, event.id)
+        event.image = image
+        session.commit()
 
-@router.get("/{event_id}", include_in_schema=False)
-async def eventview(request: Request, event_id: int,
+    messages = get_messages(session, event, uninvited_contacts)
+    return RedirectResponse(
+        router.url_path_for('eventview', event_id=event.id) +
+        f'?messages={"---".join(messages)}',
+        status_code=status.HTTP_302_FOUND)
+
+
+def process_image(img: bytes, event_id: int, img_height: int = HEIGHT) -> str:
+    """Resized and saves picture according to required height,
+    and keeps aspect ratio"""
+    image = Image.open(io.BytesIO(img))
+    width, height = image.size
+    height_to_req_height = img_height / float(height)
+    new_width = int(float(width) * float(height_to_req_height))
+    resized = image.resize((new_width, img_height), Image.ANTIALIAS)
+    file_name = f'{event_id}{PICTURE_EXTENSION}'
+    resized.save(f'{MEDIA_PATH}/{file_name}')
+    return file_name
```

Comment:
how do you check for malicious uploads?

Reply:
I limit the file types to be uploaded to png or jpg only.
Something similar also exists in the project - uploading a profile photo :)

Reply:
(you can see this in the html itself BTW not in the area you marked)

Reply:
Also in that section, I don't see any check for malicious uploads, although it is limited by size, it can still be malicious. I suggest that you check for that.

Reply:
So after digging a little in stackoverflow in the internet - 
The potentially malicious data in these types of file could only be in the metadata.
However, the fact that I open the image using PIL and then resave the picture should remove the metadata which should eliminate this issue.

In any case I added some code in this process_image function that should also manually remove the exif (the metadata) and this should assure that only the image data itself is saved :)